pandas/core/reshape/merge.py

```diff
@@ -552,8 +552,8 @@ class _MergeOperation:
 
     def __init__(
         self,
-        left: FrameOrSeries,
-        right: FrameOrSeries,
+        left: "Union[Series, DataFrame]",
+        right: "Union[Series, DataFrame]",
```

Comment:
needs quotes?

Reply:
"Union[Series, DataFrame]" -> Union["Series", "DataFrame"]

I think is clearer. not sure if we have a preferred style here. @WillAyd ?

Reply:
I agree that having the quotes inside the union would be nicer, but doing so gave flake8 complaints about Series being unused.

Reply:
adding `# noqa: F401` to the imports inside the `TYPE_CHECKING` block I think is acceptable and not uncommon in this case.

Reply:
updated